lib/biokbase/catalog/db.py

```diff
@@ -120,18 +125,69 @@ class MongoCatalogDBI:
 
     def __init__(self, mongo_host, mongo_db, mongo_user, mongo_psswd, mongo_authMechanism):
 
-        # create the client
-        self.mongo = MongoClient('mongodb://' + mongo_host)
+        self.mongo_host = mongo_host
+        self.mongo_db = mongo_db
+        self.mongo_user = mongo_user
+        self.mongo_psswd = mongo_psswd
+        self.mongo_authMechanism = mongo_authMechanism
 
-        # Try to authenticate, will throw an exception if the user/psswd is not valid for the db
-        # the pymongo docs say authenticate() is deprecated, but testing putting auth in
-        # the MongoClient call failed
-        # to do: add authMechanism as an argument
-        if (mongo_user and mongo_psswd):
-            self.mongo[mongo_db].authenticate(mongo_user, mongo_psswd, mechanism=mongo_authMechanism)
+        self.mongo_client = None
 
-        # Grab a handle to the database and collections
-        self.db = self.mongo[mongo_db]
+        # Initialize mongo client
+        mongo_client = self._initialize_mongo_client()
+
+        # Check the db schema
+        self.check_db_schema(mongo_client)
+
+        # Create indexes
+        self._create_indexes(mongo_client)
+
+        # Close the MongoDB client manually before forking
+        mongo_client.close()
+        print("MongoDB client closed.")
+
+
+    def _initialize_mongo_client(self):
+        """Initialize MongoDB client."""
+        # Use the lock to ensure only one thread initializes the mongo client at a time
```

Comment:
I think this comment is in the wrong place

Reply:
👍

Reply:
:+1: